src/main/scala/extracells/part/PartFluidExport.java

```diff
@@ -71,7 +73,21 @@ public TickRateModulation doWork(int rate, int TicksSinceLastCall) {
 					stack.setStackSize(filled);
 					stack = this.extractFluid(stack, Actionable.MODULATE);
 					if (stack != null && stack.getStackSize() > 0) {
-						facingTank.fill(this.getSide().getOpposite(), stack.getFluidStack(), true);
+						int actuallyFilled = facingTank.fill(this.getSide().getOpposite(), stack.getFluidStack(), true);
+						if (actuallyFilled < stack.getStackSize()) {
+							// try to return to AE
+							int toReturn = (int)stack.getStackSize() - actuallyFilled;
+							IAEFluidStack returned = injectFluid(AEApi.instance().storage().createFluidStack(new FluidStack(fluid, toReturn)), Actionable.MODULATE);
+							if (returned != null) {
```

Comment:
Don't think it is actually possible, since the only threaded access to the AE2 storage I'm aware of is done in crafting jobs and they operate on prefetched crafting CPU caches only. But it will certainly do no harm to double check.